File: 3.1P/Clock/Clock/Clock.cs
using System;
using System.Timers;
using System.Text;


namespace ClockTask
{
    public class Clock
    {
        //Fields
        private Counter seconds;
        private Counter minutes;
        private Counter hours;
        // Constructor
        public Clock()
        {
            seconds = new Counter("seconds");
            minutes = new Counter("minutes");
            hours = new Counter("hours");
        }
        // Functions
        public void Tick()
        {
            
            seconds.Increment();
            if (seconds.Ticks == 60)
            {
                seconds.Reset();
                minutes.Increment();
            }
            
            if (minutes.Ticks == 60)
            {
                minutes.Reset();
                hours.Increment();
            }
            
            if (hours.Ticks == 24)
            {
                Reset();
            }
        }

        public void Reset()
        {
            
            seconds.Reset();
            minutes.Reset();
            hours.Reset();
        }
       // Property
        public string Time
        {
            get
            {
                return hours.Ticks.ToString("D2") + ":" + minutes.Ticks.ToString("D2") + ":" + seconds.Ticks.ToString("D2");
            }
        }
    }
}

File: 11.1P/Clock.py
# Import the Counter from Counter
from Counter import Counter

# Inherit from Counter
class Clock:

    def __init__(self):
        self.seconds = Counter(name="seconds")
        self.minutes = Counter(name="minutes")
        self.hours = Counter(name="hours")
    # Function
    def Tick(self):
        self.seconds.Increment()
        if self.seconds.Ticks == 60:
            self.seconds.Reset()
            self.minutes.Increment()

        if self.minutes.Ticks == 60:
            self.minutes.Reset()
            self.hours.Ticks

        if self.hours.Ticks == 24:
            self.hours.Reset()
    def Reset(self):
        self.seconds.Reset()
        self.minutes.Reset()
        self.hours.Reset()
    # Property
    @property
    def Time(self):
        return f"{str(self.hours.Ticks).zfill(2)} : {str(self.minutes.Ticks).zfill(2)} : {str(self.seconds.Ticks).zfill(2)}"
